src/SSVOpenHexagon/Core/HGGraphics.cpp

```diff
@@ -147,14 +150,27 @@ void HexagonGame::draw()
 
     if(Config::get3D())
     {
-        const float depth(styleData._3dDepth);
+        const float aboveMain(-styleData._3dLayerOffset - 1);
+        const bool renderAbove(aboveMain > 0 && !styleData._3dMainOnTop);
+        const float layerOffset(renderAbove ? -1.f : styleData._3dLayerOffset);
+        const float depth(styleData._3dDepth - renderAbove * aboveMain);
```

Comment:
Some comments would be appreciated here, I don't find this code obvious, especially what `aboveMain` means and the role of `layerOffset`. Also maybe rename the variables a bit to be more meaningful, like `mustRender3DAboveMain` instead of `renderAbove`?